### PR TITLE
Optimizations

### PR DESCRIPTION
**Pull request by steelbrain**
Nothing new. Just optimizations to the existing codebase.

### CODE REVIEW

File: lib/linter.coffee [hunk 1]
```diff
@@ -70,12 +70,8 @@ class Linter
     @_statCache = new Map()
 
   observeOption: (prefix, option) ->
-    callback = @updateOption.bind(this, prefix, option)
-    @subscriptions.add atom.config.observe "#{prefix}.#{option}", callback
-
-  updateOption: (prefix, option) =>
```
Comment (AsaAyers) [inline lib/linter.coffee]: This change will break 3 different linters.

File: lib/linter.coffee [hunk 2]
```diff
@@ -132,11 +128,6 @@ class Linter
   getReportFilePath: (filePath) ->
     path.join(path.dirname(filePath), @reportFilePath)
 
-  # Private: Provide the node executable path for use when executing a node
-  #          linter
-  getNodeExecutablePath: ->
```
Comment (AsaAyers) [inline lib/linter.coffee]: This one breaks two more

File: package.json [hunk 3]
```diff
@@ -7,7 +7,7 @@
   "repository": "https://github.com/AtomLinter/linter-plus",
   "license": "MIT",
   "scripts": {
-    "lint": "./node_modules/.bin/coffeelint lib -c coffeelint.json"
+    "lint": "coffeelint lib -c coffeelint.json"
```
Comment (AsaAyers) [inline package.json]: You don't need `-c coffeelint.json`. coffeelint will pick it up automatically

**Comment by AsaAyers** [comment]
Considering the whole point of that file is to maintain compatibility as we transition all the linters you really shouldn't be making changes to it.

**Comment by steelbrain** [comment]
Man, I got one word for all the linters running on that previous API "dinosaurs". The API and still wasn't really thought out, was it? :smiley:

**Comment by AsaAyers** [comment]
Who cares? Once something is in the wild like this you have to support it until you are able to transition it to whatever is new. This is part of managing a project.

**Comment by AsaAyers** [comment]
Don't worry, in a year or two some kid with no experience in this codebase will come along and call your API a  "dinosaur" and "not really thought out".

**Comment by steelbrain** [comment]
Impossible. 'cause there isn't **any** known limitation to the linter-plus API. and there were a **lot** of known limitations when the linter API rolled out. and I am not a kid with **no experience**, You know, you should go through my list of repositories sometimes.

**Comment by AsaAyers** [comment]
HA HA HA HA HA HA HA

I said no experience with this codebase. Am I wrong and you have extensive experience and know the history of how AtomLinter/Linter grew to where it is today?

**Comment by AsaAyers** [comment]
As developers virtually every time we inherit a project and aren't working with the person who made it the lie we tell our selves was "He didn't know what he was doing, but I do!". The fact is we all make the best decisions we know how to given what we know today but history is rarely kind to such decisions. When you move on someone will pick up this project and say "He didn't know what he was doing, but I do!".

**Comment by steelbrain** [comment]
You are right about that. I have no experience with this codebase, and have a very weak understanding of how it worked back then, the process spawning part especially. Maybe because the struct was so well crafted that it takes a string containing different arguments for the process and then launches a separate process for it even if it's a node js thing, and splits the string arguments into array again and then launches a buffered process (Why did they use buffered process again?, I have been using normal child_process.exec without a problem for more than a year) and then copies the file into a separate temp file and then lints it, instead of just linting the original file or passing the file to the process via stdin and then shows the errors with a format of `CSS - CSS - Unterminated statement` where even a child knows that double CSS is redundant.

and My friend, Mr Ayers, I am the biggest supporter of revolutions that you would ever know, It's just that there's a difference when people claim stuff. For example According to hitler, killing all the jews was the best act ever. But all of us oppose it, don't we?

**Comment by steelbrain** [comment]
Anyways, I've pushed the package.json updates to the master. :smiley:

**Comment by AsaAyers** [comment]
http://en.wikipedia.org/wiki/Godwin%27s_law

**Comment by steelbrain** [comment]
HAHAHAHAHA

**Pull request closed without merging**